corpus/convert.py

```diff
@@ -326,3 +330,69 @@ def run(self):
                     segment.fullname() in segments_whitelist
                 ):
                     f.write(segment.orth + "\n")
+
+
+class CorpusToRawWavHDFJob(Job):
+    """
+    Gets audio files from a Bliss corpus and stores them as HDF file
+    compatible with the returnn HDFDataset
+    """
+
+    def __init__(
+        self,
+        bliss_corpus: tk.Path,
+        segment_file: Optional[tk.Path] = None,
+        returnn_root: Optional[tk.Path] = None,
+    ):
+        """
+
+        :param bliss_corpus: Bliss corpus to read segments and audio files from
+        :param segment_file: segment file that lists allowed segments
+        :param returnn_root: returnn repository
+        """
+        self.set_vis_name("Dump audio to HDF")
+
+        self.bliss_corpus = bliss_corpus
+        self.segment_file = segment_file
+        self.returnn_root = returnn_root
+
+        self.out_hdf = self.output_path("audio.hdf")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        SimpleHDFWriter = get_returnn_simple_hdf_writer(self.returnn_root.get_path())
+
+        c = corpus.Corpus()
+        c.load(self.bliss_corpus.get_path())
+
+        if self.segment_file:
+            with uopen(self.segment_file, "rt") as f:
+                segments_whitelist = set(
+                    l.strip() for l in f.readlines() if len(l.strip()) > 0
+                )
+        else:
+            segments_whitelist = None
+
+        out_hdf = SimpleHDFWriter(filename=self.out_hdf, dim=1)
+
+        for segment in c.segments():
+            if (not segments_whitelist) or (segment.fullname() in segments_whitelist):
+                audio_file = segment.recording.audio
+                audio = wave.open(audio_file)
```

Comment:
I think at other places, we rather use `soundfile` which is also in the `requirements.txt`. This would be more consistent.

Reply:
good point. I'll try to rewrite this.

Reply:
Not sure how much of the processing time is consumed by loading, but you might also rewrite the code a bit to load the audio only once for each recording and then iterate over the segments. In case you have a corpus with many segments per recording, this might be beneficial.

Reply:
That's true. I think the first case is more likely, but I'm fine with keeping the logic as is, too.